mesonbuild/ast/introspection.py

```diff
@@ -124,6 +124,10 @@ def func_project(self, node: BaseNode, args: T.List[TYPE_nvar], kwargs: T.Dict[s
 
         self.coredata.set_options(options)
         self.func_add_languages(None, proj_langs, None)
+        for kw in ['languages', 'native_languages']:
+            if kw in kwargs:
+                v = [kwargs.get(kw)]  # type: T.List[TYPE_nvar]
+                self.func_add_languages(None, v, None)
```

Comment:
`func_add_languages` currently iterates through both `MachineChoice.BUILD`, and `MachineChoice.HOST`. We should give `func_add_languages` a `for_machine` argument instead.

Reply:
We can't (simply) change the signature of `func_add_languages()`, because it's invoked by the parser function dispatcher.  Certainly an internal `add_languages()` method could be added.

I don't know what use (if any) the introspector makes of the language information.  At this remove in time I no longer know what problem I thought adding this was fixing (reverting it doesn't make any tests fail, now).

Reply:
> Certainly an internal `add_languages()` method could be added.

That sounds good.

> I don't know what use (if any) the introspector makes of the language information.

Even if it does none, I think using a helper function is still good. In that case, it avoids a gotcha going forward if we do try to make it use that info.

Reply:
I have factored this out as a helper function, `_add_languages()` as you suggest.